containers/podman-desktop
# feat: Add rename image button

### What does this PR do?

Adds the ability to rename images on the Image List and the Image Details pages.

**Edit Image Menu Item**
![image](https://github.com/containers/podman-desktop/assets/4624530/81c54473-23c5-4b9f-ab1a-be4d02c96f73)

**Edit Image Icon on Image Details page**
<img width="732" alt="image" src="https://github.com/containers/podman-desktop/assets/4624530/daa80f73-f3d5-4b3f-a803-d74c4e33335d">

**Modal**

![image](https://github.com/containers/podman-desktop/assets/4624530/82c64cc8-de60-4a83-8249-5c204e5da6e8)
 
**Error messages** (and placeholder text)

![image](https://github.com/containers/podman-desktop/assets/4624530/ac18c058-d452-4086-851d-e7d3c0f9d819)


### What issues does this PR fix or reference?



Closes: #2269 

### How to test this PR?



1. Pull an image
2. On the image list page click the kebab and then select "Rename Image"
   - On the details page select the "Rename Image" icon 
4. Fill out the form and click "Save"
5. See the new tag listed in the images list

<details>
<summary>Old Description</summary>

### What does this PR do?

Adds the ability to add new tags to images on the Image List and the Image Details pages.

### Screenshot/screencast of this PR



**Rename Image Menu Item**
<img width="695" alt="Screenshot 2023-05-27 at 1 41 28 PM" src="https://github.com/containers/podman-desktop/assets/4624530/d1d7f612-6b91-499c-9a2e-76882efdfb8e">

**Rename Image Icon on Image Details page**
<img width="732" alt="image" src="https://github.com/containers/podman-desktop/assets/4624530/daa80f73-f3d5-4b3f-a803-d74c4e33335d">

**Modal for both list and details page view**

![image](https://github.com/containers/podman-desktop/assets/4624530/94a2d7ca-36fe-49a7-bd63-58ed7cf4555f)


### What issues does this PR fix or reference?



Closes: #2269 

### How to test this PR?



1. Pull or build an image
2. On the image list page click the kebab and then select "Rename Image"
   - On the details page select the "Rename Image" icon 
4. Fill out the form and click "Add Image Tag"
5. See the new tag listed in the images list

### ToDo

 - [x] Create Modal
 - [x] Add rename/tag container-registry methods
 - [x] Rename/Add tag from Image List page
 - [x] Rename/Add tag from Image Details page
 - [x] Change "Add Image Tag" button to "Done" when the process is complete
 ~- [ ] (optional?) show terminal output on the Modal (similar to the PushImageModal)~ No output in the terminal

</details>
